APE9.rst

```diff
@@ -0,0 +1,273 @@
+for slicing based on world coordinates, but this will not be part of the initial
+implementation.
+
+NDData does not support mathematical operations. Utilities will be provided to aid
+in how metadata items are to be propagated when the user merges NDData instances.
+
+Mask Object
++++++++++++
+
+The mask attribute now will be an instance of a Mask Object (but to support
+backward compatibilty, the attribute must have a different name than mask since
+the current implementation treats that attribute as a boolean array (or any
+object that mimics one). [An alternative option is to assume that the mask
+attribute of NDArr is a callable whereas the mask attribute of NDData represents
+a boolean array; this would be workable, but also likely confusing]
+This APE proposes that the new attribute have the name: lone_ranger. 
+Well, perhaps dmask.
+
+As objects, masks may take many internal forms. For example, they may be 
+simple boolean arrays, but they may also have many bitplanes or even be
+computed masks. The only requirement is that this object implement a
+callable interface such that it returns a boolean array when called with
+no positional or keyword arguments. These masks may implement any number
+of customizations to the callable (e.g., optional keyword arguments) as 
+well as methods to set the default behavior for the callable (i.e., when
+called with no arguments or keywords).
+The returned boolean array must be broadcastable to the underlying data array
+(a scalar boolean is permitted).
+
+As an example for bitplane masks:
+
+- ``nda.dmask()`` returns a boolean array
+
+- ``nda.dmask.apply_flags(…)`` sets how the bit planes are to be used to generate a boolean result when ``ndd.dmask()`` is used
+
+- ``nda.dmask(select=nda.dmask.bitplanes['saturated'] + nda.dmask.bitplanes['dead_pixels'])``
+
+Binary Operations on Masks
+++++++++++++++++++++++++++
+
+When applying binary operations two different mask objects 
+objects, the result should be:
+
+
+1. The boolean result of the boolean combination of the default boolean
+representation of the two input masks.
+
+2. For bitplane masks:
+the boolean result of matched bit planes (by label) for those that have matches 
+and simple propagation of bitplanes that don’t match. An error results if the
+total required bit planes exceeds 64 (until larger int sizes are supported 
+universally in numpy or we build an interface to a higher dimensionality of
+int arrays to support more bit planes).
+
+3. For all other masks the result is a boolean array combination of the 
+default callable mask for each of the two inputs. In the case of a bit plane 
+combined with any other type of mask, the boolean will be applied to each 
+bitplane.
+
+Uncertainty (the eternal tar pit)
++++++++++++++++++++++++++++++++++
+
+The NDArr object shall have a "variance" attribute (too bad for those that
+desire standard deviation as an option; though for backward compatibility
```

Comment:
@mhvk At this point my preference is mostly based on our long-established current practice, rather than the underlying arguments -- not that it should prevent doing whatever is optimal -- I'm mainly saying that variance is fine for me and Gemini, as an interested party.